src/test/java/org/sagebionetworks/web/unitclient/widget/entity/DoiWidgetTest.java

```diff
@@ -35,101 +33,105 @@
+		testDoi.setDoiStatus(DoiStatus.IN_PROCESS);
+		doiWidget.configure(testDoi, entityId);
+		verify(mockView).showDoiInProgress();
 	}
 	
-	@SuppressWarnings("unchecked")
 	@Test
-	public void testConfigureNotFoundNonEditable() throws Exception {
-		AsyncMockStubber.callFailureWith(new NotFoundException()).when(mockSynapseClient).getEntityDoi(anyString(), anyLong(), any(AsyncCallback.class));
-		doiWidget.configure(entityId, false, null);
+	public void testConfigureNotFound() throws Exception {
+		doiWidget.configure(null, entityId);
 		verify(mockView).setVisible(false);
-	}
-
-	
-	@SuppressWarnings("unchecked")
-	@Test
-	public void testConfigureOtherError() throws Exception {
-		AsyncMockStubber.callFailureWith(new IllegalArgumentException()).when(mockSynapseClient).getEntityDoi(anyString(), anyLong(), any(AsyncCallback.class));
-		doiWidget.configure(entityId, true, null);
-		verify(mockView).showErrorMessage(anyString());
+		verify(mockView).clear();
 	}
 	
-	@SuppressWarnings("unchecked")
 	@Test
 	public void testGetDoiPrefix() throws Exception {
 		doiWidget.getDoiPrefix(mock(AsyncCallback.class));
 		verify(mockStackConfigService).getDoiPrefix(any(AsyncCallback.class));
 	}
 	
-	@SuppressWarnings("unchecked")
 	@Test
 	public void testGetDoiLink() throws Exception {
 		String prefix = "10.5072/FK2.";
 		Long version = 42l;
-		doiWidget.configure(entityId, true, version);
+		testDoi.setObjectVersion(version);
+		doiWidget.configure(testDoi, entityId);
 		String link = doiWidget.getDoi(prefix, true);
 		assertTrue(link.contains(entityId));
 		assertTrue(link.contains(version.toString()));
 		assertTrue(link.contains(prefix));
 	}
 
-	@SuppressWarnings("unchecked")
 	@Test
 	public void testGetDoiLinkNoPrefix() throws Exception {
 		String prefix = "";
 		Long version = 42l;
-		doiWidget.configure(entityId, true, version);
+		testDoi.setObjectVersion(version);
+		doiWidget.configure(testDoi, entityId);
 		String link = doiWidget.getDoi(prefix, true);
 		assertTrue(link.length() == 0);
 		link = doiWidget.getDoi(null, true);
 		assertTrue(link.length() == 0);
 	}
+
+	@Test(expected=UnsatisfiedLinkError.class)
+	public void testGetEntityDoi() {
+		// when ready
+		testDoi.setDoiStatus(DoiStatus.READY);
+		doiWidget.getEntityDoi(entityId, null);
+		verify(mockView).setVisible(false);
+		verify(mockView).clear();
+		verify(mockView).showDoiCreated(doiWidget.getDoi(testDoiPrefix, false));
+		// when in error
+		testDoi.setDoiStatus(DoiStatus.ERROR);
+		doiWidget.getEntityDoi(entityId, null);
+		verify(mockView, Mockito.times(2)).setVisible(false);
+		verify(mockView, Mockito.times(2)).clear();
+		verify(mockView).showDoiError();
+		// when in process, should throw UnsatisfiedLinkError
+		testDoi.setDoiStatus(DoiStatus.IN_PROCESS);
+		doiWidget.getEntityDoi(entityId, null);
+		verify(mockView, Mockito.times(3)).setVisible(false);
+		verify(mockView, Mockito.times(3)).clear();
+		verify(mockView).showDoiError();
```

Comment:
should not show a doi error in this case, should call view.showDoiInProgress().  But because it failed on the previous test, this verify is true.  Will break after adding the reset(view).